File: python/Basic-Algorith-Scripting/chunkyMonkey.py
'''

Write a function that splits an array (first argument) into groups the length of size (second argument) and returns them as a two-dimensional array.

'''


def chunkArrayInGroups(arr, size):
    return [arr[i:i+size] for i in range(0, len(arr), size)]

  
chunkArrayInGroups(["a", "b", "c", "d"], 2)

File: php/Basic-Algorithm-Scripting/chunkyMonkey.php
<?php

/*

Write a function that splits an array (first argument) into groups the length of size (second argument) and returns them as a two-dimensional array.

*/


function chunkArrayInGroups($arr, $size) {
    $ret = array();
    for($i=0; $i<count($arr); $i += $size) {
        $ret[] = array_slice($arr,$i, $i+$size);
    }
    return $ret;
}
  
chunkArrayInGroups(["a", "b", "c", "d"], 2);

?>